cashsvr_input_tof.c

```diff
@@ -151,29 +137,86 @@ static int cash_tof_sys_init(bool high_accuracy, int devno, int plen)
 		snprintf(sns_mode_path, plen + LEN_MODE,
 			"%s%d/set_use_case", sysfs_input_str, devno);
 
-		if (chown(sns_mode_path, uid, gid) == -1) {
+		rc = cash_set_permissions(sns_mode_path, "system", "input");
+		if (rc == -1) {
 			ALOGD("Cannot chown %s/enable_ps_sensor", sns_mode_path);
-			return 1;
-		}
-
-		fd = open(sns_mode_path, O_WRONLY);
-		if (fd < 0) {
-			ALOGD("Cannot open %s", sns_mode_path);
-			return 1;
+			free(cash_tof_enable_path);
```

Comment:
Consider setting `cash_tof_enable_path` to `NULL`, as it's checked against. Same for the `free` at line 133.

Reply:
Also, why is the global enable path freed in this error case, but not in the error cases below?